### DISCUSSION TITLE
Pybricksdev device/address serialization

### DISCUSSION THREAD

**Discussion by Novakasa** [pybricks/support#341] [Q&A] [answered]
Hi all,

So until recently, I've been using the BLE address to uniquely identify my hubs. However, it seems in a recent update, PybricksHub.connect(address_str) raises an AttributeError because it now only expects a `BLEDevice` instead of an address string. Is it discouraged to use the address as an identifier now? In my application, I want to be able to dynamically store hub identification on disk. It would be convenient to use just the address for this, but reading e.g. https://github.com/pybricks/pybricksdev/pull/18#discussion_r625168874 , it seems a Bleak `BLEDevice` will make it faster to connect to the hub. Is it a good idea to serialize this object instead? It seems for different backends this object looks different internally, so maybe just using the address might be better for cross-compatibility reasons.

**Comment by dlech** [comment]
In our next release (v3.1 firmware) we would like to add a feature to give each hub a unique name so that we can use the name instead of the address of the hub.

We try to avoid using the address because Apple does not allow accessing the address on macOS for privacy and security reasons. For that reason, we can't make generic instructions to "type your Bluetooth address here" that work for everyone. But serializing it to disk so that the hub is "remembered" sounds like a legitimate use case since it would still work on macOS.

There is a [pull request](https://github.com/pybricks/pybricksdev/pull/18) to fix `pybricksdev` to allow the `find_device()` function to take an address instead of the name but it seems to have stalled.

**Reply by Novakasa** [reply]
Thanks, that makes sense. If I ever need to support mac I will make sure to use the UUID then.
I might submit another pull request then if there are no updates on https://github.com/pybricks/pybricksdev/pull/18

**Reply by dlech** [reply]
FYI, on macOS, the `address` attribute from Bleak will return the UUID instead of the Bluetooth address so no changes should be necessary.

**Comment by dlech** [comment, marked as answer]
Please try pybricksdev v1.0.0-alpha.8.

You will probably need to do something like this now:

```python
device = await find_device(address_str)
await PybricksHub.connect(device)
```
